### PR TITLE
hw30

### PR DESCRIPTION
**Pull request by RomanBletsko**

### CODE REVIEW

File: src/components/Content/Login/Form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  const context = useContext(UserContext);
+  const { loader } = useSelector((store)=>store.login)
+  const dispatch = useDispatch();
+  const [loginError, setLoginError]= useState(false)
+
+  const handleLogin = (values)=>{
+    setLoginError(false)
+      dispatch(fetchLogin({...values}))
+      .then((responce) => context.saveAccessToken(responce.payload.token))
+      .catch(() => {setLoginError(true)});
```
Comment (roskyba) [inline src/components/Content/Login/Form/index.jsx]:
```suggestion
    setLoginError(false)
    dispatch(fetchLogin({...values}))
        .then((response) => context.saveAccessToken(response.payload.token))
        .catch(() => {setLoginError(true)});
```

File: src/components/Content/Login/Form/index.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+    
+    return(<WraperStyled>
+      {loader ? (<Loader />):(
```
Comment (roskyba) [inline src/components/Content/Login/Form/index.jsx]:
```suggestion
      { loader ? <Loader /> : (
```

File: src/components/Content/Login/Form/index.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        </FormStyled>
+        )} 
+      </Formik></>
+        
+      )}
+        
+    </WraperStyled>)
+}
+const WraperStyled = styled.div({
+  // width: "80%",
```
Comment (roskyba) [inline src/components/Content/Login/Form/index.jsx]:
```suggestion
```

File: src/components/Content/Posts/Post/index.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            <p>{name}</p>
+            <p>{year}</p>
+        </Wraper>
+    )
+}
+const Wraper = styled.div`
+height: 100px;
+background: ${(props) => {
+   return props.color
+}};
```
Comment (roskyba) [inline src/components/Content/Posts/Post/index.jsx]:
```suggestion
background: ${( props ) => props.color };
```

File: src/components/Content/Posts/index.jsx [hunk 5]
```diff
@@ -1,7 +1,91 @@
...
+import Pagination from '@mui/material/Pagination';
+import Stack from '@mui/material/Stack';
+import Loader from "../../Loader";
+import Post from "./Post";
+
 
 const Posts = () => {
-  return <div>Posts</div>;
+  const { posts,status,page,totalPages,loader } = useSelector((store) => store.posts);
+  // const [pageNumber,setPageNumber] = useState(1)
```
Comment (roskyba) [inline src/components/Content/Posts/index.jsx]:
```suggestion
```

File: src/components/Content/Posts/index.jsx [hunk 6]
```diff
@@ -1,7 +1,91 @@
...
+  //       if(action === 'Go to next page'){
+  //         setPageNumber( pageNumber + 1)
+  //         dispatch(fetchPosts({page:pageNumber + 1}))
+  //       }else{
+  //         setPageNumber( pageNumber - 1)
+  //         dispatch(fetchPosts({page:pageNumber - 1}))
+  //       }
+  //     }
+  //   }
+  // }
```
Comment (roskyba) [inline src/components/Content/Posts/index.jsx]:
```suggestion
```

File: src/components/Content/Posts/index.jsx [hunk 7]
```diff
@@ -1,7 +1,91 @@
...
+  </ListStyled>)}
+  <PaginationHolder>
+    <Stack spacing={2}>
+      <Pagination count={totalPages} 
+      variant="outlined" 
+      shape="rounded" 
+      onChange={handleFunc}
+      // onClick={(event)=>{
+      //   handleFunc(event.target) 
+      // }}
```
Comment (roskyba) [inline src/components/Content/Posts/index.jsx]:
```suggestion
```

File: src/components/Content/Users/SingleUser/index.jsx [hunk 8]
```diff
@@ -5,7 +5,7 @@ import "./styles.css";
 
 function SingleUser({ first, last, email, avatar }) {
   const context = useContext(ThemeContext);
-  console.log("context", context);
+  // console.log("context", context);
```
Comment (roskyba) [inline src/components/Content/Users/SingleUser/index.jsx:8]:
```suggestion
```

File: src/components/Tabs/index.jsx [hunk 9]
```diff
@@ -5,7 +5,7 @@ import "./styles.css";
 
 const Tabs = () => {
   const context = useContext(ThemeContext);
-  console.log("context = ", context);
+  // console.log("context = ", context);
```
Comment (roskyba) [inline src/components/Tabs/index.jsx:8]:
```suggestion
```

File: src/context/themeContext.jsx [hunk 10]
```diff
@@ -2,4 +2,4 @@ import { createContext } from "react";
 
 export const ThemeContext = createContext(null);
 
-console.log("ThemeContext =", ThemeContext);
+// console.log("ThemeContext =", ThemeContext);
```
Comment (roskyba) [inline src/context/themeContext.jsx:5]:
```suggestion
```

**Review by roskyba — APPROVED** [review]
Good work!

File: src/components/Content/Login/Form/index.jsx (hunk 1 above)
Comment (RomanBletsko) [reply src/components/Content/Login/Form/index.jsx]: fixed

File: src/components/Content/Login/Form/index.jsx (hunk 2 above)
Comment (RomanBletsko) [reply src/components/Content/Login/Form/index.jsx]: fixed

File: src/components/Content/Login/Form/index.jsx (hunk 3 above)
Comment (RomanBletsko) [reply src/components/Content/Login/Form/index.jsx]: fixed

File: src/components/Content/Posts/Post/index.jsx (hunk 4 above)
Comment (RomanBletsko) [reply src/components/Content/Posts/Post/index.jsx]: fixed

File: src/components/Content/Posts/index.jsx (hunk 5 above)
Comment (RomanBletsko) [reply src/components/Content/Posts/index.jsx]: fixed

File: src/components/Content/Posts/index.jsx (hunk 6 above)
Comment (RomanBletsko) [reply src/components/Content/Posts/index.jsx]: fixed

File: src/components/Content/Posts/index.jsx (hunk 7 above)
Comment (RomanBletsko) [reply src/components/Content/Posts/index.jsx]: fixed

File: src/components/Content/Users/SingleUser/index.jsx (hunk 8 above)
Comment (RomanBletsko) [reply src/components/Content/Users/SingleUser/index.jsx:8]: fixed

File: src/components/Tabs/index.jsx (hunk 9 above)
Comment (RomanBletsko) [reply src/components/Tabs/index.jsx:8]: fixed

File: src/context/themeContext.jsx (hunk 10 above)
Comment (RomanBletsko) [reply src/context/themeContext.jsx:5]: fixed

**Pull request merged**